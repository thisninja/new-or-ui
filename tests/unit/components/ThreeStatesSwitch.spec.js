import Component from '@/components/ThreeStatesSwitch/main.vue';

test('renders correctly', () => {
  const { element } = shallowMount(Component, {
    propsData: { value: true }
  });
  expect(element).toMatchSnapshot();
});

test('checked=SWITCH_PARTIAL', () => {
  const { element } = shallowMount(Component, {
    propsData: { value: null }
  });
  expect(element.classList.contains('cl-three-states-switch')).toBe(true);
});
