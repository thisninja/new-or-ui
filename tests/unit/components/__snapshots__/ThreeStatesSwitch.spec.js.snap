// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="cl-three-states-switch"
  style="width: 52px; height: 24px; border-radius: 4px; justify-content: space-around; background-color: rgb(57, 191, 102); color: white;"
>
  <div
    class="text-smaller"
  >
    
        on
    
  </div>
   
  <div
    class="square"
    style="width: 16px; height: 16px;"
  />
   
  <!---->
</div>
`;
